### PR TITLE
Very basic shooter functionality (setRPM, and RT -> lots of volts)

### PR DESCRIPTION
**Pull request by ChaoticChaosTheGreat**
No idea if this works.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    if (RobotBase.isReal()) {
+      configureRealHardware();
+    }
+
+    System.out.println("Shooter initialized");
+  }
+
+  private void configureRealHardware() {
+    shooterMotor = new TalonFX(ShooterConstants.kShooterMotorID);
+    shooterMotor.setInverted(false);
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:42]: what's this for?

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.ShooterConstants;
+
+public class Shooter extends SubsystemBase {
+
+  final VelocityVoltage m_request = new VelocityVoltage(0).withSlot(0);
+
+  SimpleMotorFeedforward shooterFeedforward =
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:24]: private or public

**Review by realjoshuau — COMMENTED** [review]
LGTM; just fix PID constants & test IRL and i'll approve

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  public double calculateFeedForward(double velocity, double acceleration) {
+    return shooterFeedforward.calculate(velocity, acceleration);
+  }
+
+  public void setInputVoltage(double voltage) {
+    shooterMotor.setVoltage(MathUtil.clamp(voltage, -12, 12));
+  }
+
+  public void setVelocity(double velocity) {
+    shooterMotor.setControl(m_request.withVelocity(velocity).withFeedForward(0.5));
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:65]: what's the .5 for

File: src/main/java/frc/robot/subsystems/shooter/commands/SetShooterVoltage.java
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/shooter/commands/SetShooterVoltage.java:1]: What's the point of this when we have the RPM commands

Comment (realjoshuau) [reply src/main/java/frc/robot/subsystems/shooter/commands/SetShooterVoltage.java:1]: For testing

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java (hunk 3 above)
Comment (ChaoticChaosTheGreat) [reply src/main/java/frc/robot/subsystems/shooter/Shooter.java:65]: .5 multiplier

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java (hunk 2 above)
Comment (ChaoticChaosTheGreat) [reply src/main/java/frc/robot/subsystems/shooter/Shooter.java:24]: private i suppose

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java (hunk 1 above)
Comment (ChaoticChaosTheGreat) [reply src/main/java/frc/robot/subsystems/shooter/Shooter.java:42]: Configure shooter motor? I left the inverted line in there so in the future if it does need to be inverted I can just change it to true.

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.ShooterConstants;
+
+public class Shooter extends SubsystemBase {
+
+  final VelocityVoltage m_request = new VelocityVoltage(0).withEnableFOC(true).withSlot(0);
+
+  SimpleMotorFeedforward shooterFeedforward =
```
Comment (realjoshuau) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:24]: Why??

**Review by realjoshuau — CHANGES_REQUESTED** [review]
Remove unnecessary feed forward.

**Review by realjoshuau — CHANGES_REQUESTED** [review]
`Cannot invoke "com.ctre.phoenix6.hardware.TalonFX.getMotorVoltage()" because "this.shooterMotor" is null`

@ThatXliner Sim is broken due to TalonFX being null

**Review by realjoshuau — APPROVED** [review]
lgtm

**Pull request merged**